Arch arm v8m mpu fix buffer validate func

ioannisg opened the pull request:
- Fixing a bug in _mpu_buffer_validate for ARMv8-M MCUs with TrustZone support. The bug is assigned priority low, given that there are a couple of such boards in the tree (musca_a and nrf9160_pca10090).
- Assigning Code owner for ARMv8-M CMSE module

ioannisg committed:
CODEOWNERS: add code-owner for cortex-m/cmse module

Code-owner for the ARM/Cortex-M/CMSE module (source
and include directories).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: mpu: improve documentation of ARMv8-M _mpu_buffer_validate

Enhance the documentation for internal ARMv8-M MPU
function _mpu_buffer_validate().

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

MaureenHelm commented on arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h:
```diff
@@ -238,11 +238,11 @@ static inline int _is_enabled_region(u32_t index)
- * Therefore, the function perform a second check, in case the fast address
- * range check fails, which is based on MPU only.
+ * Therefore, the function performs a second check, which is based on MPU only,
+ * in case the fast address range check fails.
```
I think these comment edits belong with the previous doc commit

ioannisg replied:
Yeap, my bad, will fix that :)

ioannisg replied:
This is now fixed.

MaureenHelm reviewed (changes requested):
Just one nit, otherwise looks fine

ioannisg committed:
arch: arm: mpu: fix _mpu_buffer_validate() implementation

In this commit we fix the implementation of internal function
_mpu_buffer_validate(), so it can work in the presence of a
security attribution unit (SAU, or IDAU). The fast validation
based on the CMSE address range check intrinsic is performed
first: if it fails, then a second, MPU-only check is performed
for ARMV8-M platforms that have TEE capabilities (i.e. SAU/IDAU
units).

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg reviewed (commented):
Fixed the issue with the contents of each commit, @MaureenHelm, please, re-visit.

@galak : this is a (agree, a minor) bug in ARMv8-M builds with or without TrustZone-M; even if we build an ARMv7-M compatible firmware, i.e. ignoring TrustZone, the cmse_address_range_check intrinsic still consults the existing SAU/IDAU to assess the RW permissions. So, without that patch, large objects that span multiple IDAU regions (e.g. 8 KB for nRF9160) would not be validated as read-writeable, even if MPU allowed that.

Regarding your Slack question, yes, CMSE is used in any ARMv8-M build with MPU, regardless of Trusted Execution.

marc-hb commented on CODEOWNERS:
```diff
@@ -9,6 +9,8 @@
 /arch/arc/                                @vonhust @ruuddw
 /arch/arm/                                @MaureenHelm @galak
+/arch/arm/core/cortex_m/cmse/             @ioannisg
+/arch/arm/include/cortex_m/cmse/          @ioannisg
```
cc: @nashif 
There's no `/arch/arm/include/cortex_m/cmse/` directory , did you mean `/arch/arm/include/cortex_m/cmse.h` ?

As reported by the new and fixed Codeowners check just merged in PR https://github.com/zephyrproject-rtos/ci-tools/pull/54

marc-hb replied:
Forgot to explain the urgency of this issue sorry: now anyone trying to add a new file anywhere will experience a failed Codeowners check in github. The "quickfix" is to remove this line; is there a better fix?

ioannisg replied:
Hi, sorry for this. Just replace the line as you proposed, above (with cmse.h)
